astropy/cosmology/tests/test_connect.py

```diff
@@ -0,0 +1,130 @@
+# Licensed under a 3-clause BSD style license - see LICENSE.rst
+
+import copy
+import os
+import json
+
+import pytest
+
+import numpy as np
+
+import astropy.units as u
+from astropy import cosmology
+from astropy.cosmology import Cosmology
+from astropy.cosmology.connect import CosmologyRead
+from astropy.cosmology.core import _COSMOLOGY_CLASSES, Cosmology
+from astropy.io import registry as io_registry
+from astropy.utils.exceptions import AstropyUserWarning
+
+cosmo_instances = cosmology.parameters.available
+save_formats = ["json"]
+
+
+# make a common directory for reading / writing cosmologies
```

Comment:
Any reason you do not just use `tmpdir` in the relevant methods?

Reply:
Only that I didn't know it was a builtin pytest fixture. Thought I had to make my own.